pin/pin.go

```diff
@@ -519,7 +520,9 @@ func (p *pinner) PinWithMode(c *cid.Cid, mode PinMode) {
 	}
 }
 
-func hasChild(ds mdag.LinkService, root *cid.Cid, child *cid.Cid) (bool, error) {
+// hasChild recursively looks for a Cid among the children of a root Cid.
+// The visit function can be used to shortcut already-visited branches.
+func hasChild(ds mdag.LinkService, root *cid.Cid, child *cid.Cid, visit func(*cid.Cid) bool) (bool, error) {
```

Comment:
Because it creates an extra indirection which does not help to figure out what this is doing any more than having the definition directly as parameter. Moreover, this is the only parameter which uses that in the module, it's not hard to read and it is called from a single place which clearly shows that expected usage is to pass cid.Set.Visit function.